### ISSUE TITLE
git-clone tutorial failing

### ISSUE THREAD

**Issue by jace-ys** [tektoncd/catalog#224]
# Expected Behavior

Running the [git-clone tutorial](https://github.com/tektoncd/catalog/tree/master/git) should see the PipelineRun for `cat-pipeline-readme-run` succeed.

# Actual Behavior

The actual repository cloning succeeds, but an unknown `$(results.commit.path)` causes the PipelineRun to fail instead.

Logs from PipelineRun:
```sh
+ CHECKOUT_DIR=/workspace/output/pipeline-checkout
+ '[[' false '==' true ]]
+ /ko-app/git-init -url https://github.com/tektoncd/pipeline.git -revision master -path /workspace/output/pipeline-checkout -sslVerify true -submodules true -depth 1
{"level":"info","ts":1584439500.8515205,"caller":"git/git.go:102","msg":"Successfully cloned https://github.com/tektoncd/pipeline.git @ master in path /workspace/output/pipeline-checkout"}
{"level":"warn","ts":1584439500.8516173,"caller":"git/git.go:149","msg":"Unexpected error: creating symlink: symlink /tekton/home/.ssh /root/.ssh: file exists"}
{"level":"info","ts":1584439500.8988461,"caller":"git/git.go:130","msg":"Successfully initialized and updated submodules in path /workspace/output/pipeline-checkout"}
+ cd /workspace/output/pipeline-checkout
+ tr -d '\n'
+ git rev-parse HEAD
+ RESULT_SHA=eb5d42d0cf4a1b0abde2a2d195d121123f5c7662
+ EXIT_CODE=0
+ '[' 0 '!=' 0 ]
+ results.commit
/tekton/scripts/script-0-c4sqd: line 40: results.commit.path: not found
/tekton/scripts/script-0-c4sqd: line 40: can't create : nonexistent directory
+ echo -n eb5d42d0cf4a1b0abde2a2d195d121123f5c7662
```

I see that there's some work going on for the `git-clone` script in https://github.com/tektoncd/catalog/pull/188. Not sure if that will address this issue?

# Steps to Reproduce the Problem

1. Copy `git-clone.yaml` locally
2. Apply the manifests provided in the tutorial: `kubectl --namespace tekton-pipelines -f [files]`
3. See log output using `tkn --namespace tekton-pipelines pr logs cat-pipeline-readme-run`

# Additional Info

- Kubernetes version:

Kubernetes on GKE

```
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.2", GitCommit:"59603c6e503c87169aea6106f57b9f242f64df89", GitTreeState:"clean", BuildDate:"2020-01-23T14:21:36Z", GoVersion:"go1.13.6", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"14+", GitVersion:"v1.14.10-gke.17", GitCommit:"bdceba0734835c6cb1acbd1c447caf17d8613b44", GitTreeState:"clean", BuildDate:"2020-01-17T23:10:13Z", GoVersion:"go1.12.12b4", Compiler:"gc", Platform:"linux/amd64"}
```

- Tekton Pipeline version:

```
Client version: 0.8.0
Pipeline version: v0.10.1
```

- [Tekton Dashboard](https://github.com/tektoncd/dashboard) version: v0.5.2

**Comment by sbwsg** [comment]
Thanks for the issue report and apologies for confusion caused by lack of documentation here.  The `git-clone` Task was recently updated to support a new feature of Tekton Pipelines called `"resource results"`.  This feature allows Tasks, like `git-clone`, to report on data from the execution (like which git commit SHA was cloned).  That feature only got released in Pipelines 0.11.

In order to use the `git-clone` Task, the best option would be to update to Tekton Pipelines 0.11-rc2 (see [the releases page](https://github.com/tektoncd/pipeline/releases))

If it's absolutely not possible for you to upgrade your Tekton version currently then you can check out the `git-clone` Task from an older version of the catalog.  e.g. https://github.com/tektoncd/catalog/commit/bac8db44a026f3da7294be4d8e5fddad6d8598e0.  Make sure to re-read the README of the Task.  The configuration (params, volumes, etc) were considerably different in that prior version.

**Comment by jace-ys** [comment]
> Thanks for the issue report and apologies for confusion caused by lack of documentation here. The `git-clone` Task was recently updated to support a new feature of Tekton Pipelines called `"resource results"`. This feature allows Tasks, like `git-clone`, to report on data from the execution (like which git commit SHA was cloned). That feature only got released in Pipelines 0.11.
> 
> In order to use the `git-clone` Task, the best option would be to update to Tekton Pipelines 0.11-rc2 (see [the releases page](https://github.com/tektoncd/pipeline/releases))
> 
> If it's absolutely not possible for you to upgrade your Tekton version currently then you can check out the `git-clone` Task from an older version of the catalog. e.g. [bac8db4](https://github.com/tektoncd/catalog/commit/bac8db44a026f3da7294be4d8e5fddad6d8598e0). Make sure to re-read the README of the Task. The configuration (params, volumes, etc) were considerably different in that prior version.

I see, should have seen that coming! Unfortunately I don't operate the Kubernetes cluster, so I won't be able to upgrade it for Pipeline 0.11. Will have a look at the earlier docs, thanks for pointing them out!

Feel free to close this issue.
